### ISSUE TITLE
Not seeing validation errors when using ember-changeset-validation

### ISSUE THREAD

**Issue by hvgotcodes** [martndemus/ember-form-for#113]
In my app I add the following to a template

```
 {{#form-for (changeset (hash name="blah") validation)  as |f|}}
         {{f.text-field "name"}}
         {{f.submit "work"}}
{{/form-for}}
```

`validation` is a property on the controller that looks like

```
validation: {
        name: [
            validateFormat({type: 'email'}),
        ],
    },
```

As soon as I type in the text field I see a console error

`form-field.js:118 Uncaught TypeError: errors.forEach is not a function`

If I pause in the debugger, I see `errors` is an object literal (not an array) with a `validation` array in it, which contains an error appropriate for state of the input (value is not an email address). 

If I transform the `errors` object literal into an array, I get no console errors, but don't see any validation information in the form.  I used ember inspector to inspect the form-errors subcomponent of the text field, and again it looks like it is expecting arrays, and not object literals.  I stopped digging at that point.   Seems like at some point the changeset API got out of sync with what form-for expects.

**Comment by martndemus** [comment]
You should read the part about ember-changeset in the README, it's all the way at the bottom.

**Comment by hvgotcodes** [comment]
martndemus not sure what you are pointing me to.  I did follow the documentation (including the configuration of the property name to find the errors).  The problem isn't that the errors object is null, it's that its an object literal.  The form-for code that uses seems to expect an array.

**Comment by martndemus** [comment]
Are you sure you set the config to`error`, not `errors`?

**Comment by hvgotcodes** [comment]
I copied and pasted the config in the docs to my config/environment.js file. :)

This is what is in the file

```
'ember-form-for': {
            errorsProperty: 'error',
        },
```

**Comment by martndemus** [comment]
Twiddle with replication: https://ember-twiddle.com/4dc89f594487af1696f064c987f6da0f?openFiles=twiddle.json%2C

**Comment by patrick-radius** [comment]
i have exactly the same.
unfortunatly i can not figure out if the bug is introduced here or in ember-changeset-validation.

**Comment by utilityboy** [comment]
Validation errors do fire correctly when using the `errorsProperty` as @martndemus suggests.

You can see this because the `fieldHasErrorClasses` class is applied to the form field for an invalid field.

Error messages are not being displayed however because `ember-changeset` (not `ember-changeset-validations`) hangs the actual array of messages nested in `error.validation`.

@martndemus I'm not sure I can think of a nice way to reach into that object without specific branching logic to accommodate `ember-changeset`.

Also to watch out for: https://github.com/DockYard/ember-changeset/issues/100

**Comment by ndemonner** [comment]
I fixed this by adding a new config option when using `ember-changeset`. See https://github.com/SenecaSystems/ember-form-for/commit/6c04217a7abf767f6ecb1db91c8a944f5a30972d. I haven't submitted a PR for this change, because it's a hack, but it will fix the problem for people who need it now.

Probably the best way to fix this permanently would be to allow a config option called `errorPath`, defined as a function that takes a `propertyName` and returns the correct path to the error array. This would replace the `errorsProperty` present today.

**Comment by martndemus** [comment]
> One way to fix this permanently would be to allow a config option called errorPath, defined as a function that takes a propertyName and returns the correct path to the error array. This would replace the errorsProperty present today.

I like this.

**Comment by AO16** [comment]
👍 Ran into this issue as well. I like the idea above. I think that should cover use cases outside of changeset. Any one working on this? I could put something together.

**Comment by martndemus** [comment]
This issue should be sorted out by @AO16's patch, which is included in the 2.0.0-alpha.12 release
